velox/benchmarks/SimpleArithmetic.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/expression/tests/VectorFuzzer.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/Arithmetic.h"
+#include "velox/parse/ExpressionsParser.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::test;
+
+namespace {
+
+class SimpleArithmeticBenchmark
+    : public functions::test::FunctionBenchmarkBase {
+ public:
+  SimpleArithmeticBenchmark() : FunctionBenchmarkBase() {
+    registerFunction<functions::MultiplyFunction, double, double, double>(
+        {"multiply"});
+  }
+
+  void setInput(const TypePtr& inputType, const RowVectorPtr& rowVector) {
+    inputType_ = inputType;
+    rowVector_ = rowVector;
+  }
+
+  // Runs `expression` `times` times.
+  void run(const std::string& expression, size_t times) {
+    folly::BenchmarkSuspender suspender;
+    auto exprSet = compileExpression(expression, inputType_);
+    suspender.dismiss();
+
+    size_t count = 0;
+    for (auto i = 0; i < times; i++) {
+      count += evaluate(exprSet, rowVector_)->size();
+    }
+    folly::doNotOptimizeAway(count);
+  }
+
+ private:
+  TypePtr inputType_;
+  RowVectorPtr rowVector_;
+};
+
+SimpleArithmeticBenchmark benchmark;
```

Comment:
Does moving this inside the benchmarks effect the results? could it for some cases.
The state of the exec environment here is shared across iterations and across benchmarks.

Reply:
The only issue with moving would be having to somehow generate the same input dataset multiple times. Also, we need to check how the macro defines the actual function to see if it would support moving.

Reply:
oh I meant like not moving std::move , but like having an instance for every one.
yeh I agree it will recreate let revisit if we found flakiness later or incosisitency